rt/transports/http/src/main/java/org/apache/cxf/transport/http/HttpClientHTTPConduit.java

```diff
@@ -243,13 +243,18 @@ private boolean isSslTargetDifferent(URI lastURL, URI url) {
                 || !lastURL.getHost().equals(url.getHost())
                 || lastURL.getPort() != url.getPort();
     }
-    
+
     @Override
     public void close(Message msg) throws IOException {
+        OutputStream os = msg.getContent(OutputStream.class);
+        // Java 21 may hang on close, we flush stream to help close them out.
+        if (os != null && AutoCloseable.class.isAssignableFrom(HttpClient.class)) {
```

Comment:
Think it though, the `os.flush()`  my end up with IOException, should be safer to:

```

        try (OutputStream os = msg.getContent(OutputStream.class)) {
            // Java 21 may hang on close, we flush stream to help close them out.
            if (os != null && AutoCloseable.class.isAssignableFrom(HttpClient.class)) 
                 os.flush();
            } 
        }  finally {
           super.close(msg);
           msg.remove(HttpClient.class);
        }
```

Wdyt?

Reply:
Running a build with the above 
Will let you know how it tests

Reply:
NettyHttpConduit does not like it.
Trying a different impl.

Reply:
NettyHttpConduit  should not be impacted, right?